lib/fake_braintree/subscription.rb

```diff
@@ -87,7 +87,7 @@ def discounts
     def discounts_or_add_ons(discount_or_add_on)
       return [] unless discount_or_add_on.is_a?(Hash)
 
-      if discount_or_add_on['add']
+      if Array(discount_or_add_on['add']).any?
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.